src/sentry/integrations/msteams/spec.py

```diff
@@ -0,0 +1,59 @@
+from sentry import analytics
+from sentry.incidents.models.alert_rule import AlertRuleTriggerAction
+from sentry.incidents.models.incident import Incident, IncidentStatus
+from sentry.integrations.base import IntegrationProvider
+from sentry.integrations.messaging import MessagingIdentityLinkViewSet, MessagingIntegrationSpec
+from sentry.models.notificationaction import ActionService
+from sentry.rules.actions import IntegrationEventAction
+
+
+class MsTeamsMessagingSpec(MessagingIntegrationSpec):
+    @property
+    def provider_slug(self) -> str:
+        return "msteams"
```

Comment:
Should these eventually live in an enum somewhere vs just being hardcoded on the spec?

Reply:
Maybe. My problem for now is that there are too many enums already that map onto integration providers, none of which represent quite the same thing: `ActionService`, `ExternalProviders`, `ExternalProviderEnum`. I'd rather unify them than add another one.

Furthermore, I'd actually prefer not to have enums like that at all. It's a bit of an antipattern to have places in the code where we have to list every integration of a certain kind, rather than having each spec subclass stand on its own. (Imagine adding a new messaging integration. How many places do we need to update?)

I tend to think of a hardcoded return value of a property being similar in spirit to a class-level constant, in that it's declared in one place and can't be (normally) mutated. In other words, we could have written

```python
class MsTeamsMessagingSpec(MessagingIntegrationSpec):
    PROVIDER_SLUG = "msteams"
```

but referring to `MsTeamsMessagingSpec().provider_slug` is basically the same as `MsTeamsMessagingSpec.PROVIDER_SLUG`. Does it feel this way to folks, or is it just me?

Reply:
I agree that having enums for everything is an antipattern, so I wonder if these different values are things we should be consolidating some of these enums in static configs for each provider vs just placing them on the classes themselves? I guess my main concern is if we have to copy the provider slug to each distinct spec implementation for a given provider, which seems brittle.